tests/unit/slang-utils/create-parser.test.js

```diff
@@ -90,14 +90,14 @@ describe('inferLanguage', function () {
 
   for (const { description, source, version } of fixtures) {
     test(description, function () {
-      const inferredLanguage = inferLanguage(source);
-      expect(inferredLanguage.version).toEqual(version);
+      const parser = createParser(source);
+      expect(parser.version).toEqual(version);
     });
   }
 
-  test('should throw an error if there are incompatible ranges', function () {
+  test.skip('should throw an error if there are incompatible ranges', function () {
```

Comment:
this test is skipped. not sure if that was intended?

Reply:
It's intended to be skipped for now.

before we were joining pragma versions into a single string and it was straight forward to find an inconsistency, but as you pointed out, there are some edge cases where this approach leads to an error. Now we evaluate each pragma version against slang's supported versions, and now an incompatibility in the pragma versions is no different from an eager developer chasing to use the latest solidity release which might not be supported by slang just yet but with a syntax fully supported by slang.

so we might choose to remove this test as it might not be a responsibility of `prettier-solidity` or we will have to add extra steps in the version inference to double check for the validity of the pragmas.